Update the Vector documentation (#941)
k8ssandra/k8ssandra-operator: docs/content/en/tasks/monitor/vector/_index.md

title: "Using Vector with k8ssandra-operator"
linkTitle: "Vector"
weight: 6
description: "Configure Vector"
---

k8ssandra-operator provides a Vector instance that can be used to scrape logs and metrics. It's configuration is fully customizable and one can add transformers and sinks to provide the kind of solution user needs.

More information about Vector can be found in the [official documentation](https://vector.dev/docs/).

## Enabling Vector agent

> Vector agent is enabled by default for Cassandra pods. To enable Vector agent for all the other pods, you need to add a `.spec.cassandra.telemetry` section in the `K8ssandraCluster` manifest:

Reviewer: ```suggestion
Vector agent is enabled by default for Cassandra pods. To enable Vector agent for Stargate and Reaper, you need to add a `.spec.stargate.telemetry` and a `.spec.reaper.telemetry` sections respectively in the `K8ssandraCluster` manifest with `.vector.enabled: true`:
```

The example below should probably not be using `.spec.cassandra`  but rather `.spec.stargate` for example.

{"name":"org_apache_cassandra_metrics_table_bloom_filter_off_heap_memory_used","tags":{"cluster":"Weird Cluster Name","datacenter":"dc1","host":"95c50ce3-2c91-46bb-9500-536f0959241b","instance":"10.244.2.8","keyspace":"system_auth","rack":"default","table":"network_permissions"},"timestamp":"2023-02-02T10:28:08.670070700Z","kind":"absolute","gauge":{"value":0.0}}
```

As an another example, to output the logs as json to the console, one could modify the configuration to following:

```yaml
cassandra:  
  telemetry:
      vector:
        enabled: true
        components:
          sinks:
            - name: console_output
              type: console
              inputs:
                - parse_cassandra_log
              config: |
                target = "stdout"
                [sinks.console_output.encoding]
                codec = "json"
        scrapeInterval: 30s
        resources:
          requests:
            cpu: 1000m
            memory: 1Gi
          limits:

>             memory: 2Gi

Reviewer: These aren't taken into account, right?
Do we have a ticket to track making these values correctly mapped to the cassdc object (and what that mapping should be)?

Author: Well, they're read - but they apply them to the wrong place in the CassDc (they do it like the tests say.. but tests are wrong ;)). 

There's another PR, but no ticket.

Author: No, there was a ticket.. https://github.com/k8ssandra/k8ssandra-operator/issues/940
